lib/Basic/Remangler.cpp

```diff
@@ -679,6 +679,24 @@ void Remangler::mangleDependentGenericSameTypeRequirement(Node *node) {
   mangleDependentGenericParamIndex(NumMembersAndParamIdx.second);
 }
 
+void Remangler::mangleDependentGenericLayoutRequirement(Node *node) {
+  auto NumMembersAndParamIdx = mangleConstrainedType(node->getChild(0).get());
+  switch (NumMembersAndParamIdx.first) {
+    case -1: Buffer << "RL"; return; // substitution
+    case 0: Buffer << "Rl"; break;
+    case 1: Buffer << "Rm"; break;
+    default: Buffer << "RM"; break;
+  }
+  mangleDependentGenericParamIndex(NumMembersAndParamIdx.second);
+  assert(node->getChild(1)->getKind() == Node::Kind::Identifier);
+  assert(node->getChild(1)->getText().size() == 1);
+  Buffer << node->getChild(1)->getText()[0];
+  if (node->getNumChildren() >=3)
+    mangleChildNode(node, 2);
```

Comment:
Sure, working on this. I was not aware of test/Demangle/Inputs/manglings.txt until you mentioned it.